frame/contracts-registry/src/tests.rs

```diff
@@ -0,0 +1,178 @@
+	});
+}
+
+#[test]
+fn root_can_add_judge() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let acc = account("a", 1, SEED);
+
+		assert_ok!(ContractsRegistry::add_judge(origin, acc));
+	});
+}
+
+#[test]
+fn root_can_not_add_too_many_judge() {
+	new_test_ext().execute_with(|| {
+		let max_judges: u32 = <Test as Config>::MaxJudges::get();
+		for s in 0..max_judges {
+			let acc = account("a", 1, s);
+			assert_ok!(ContractsRegistry::add_judge(RawOrigin::Root.into(), acc));
+		}
+
+		let acc = account("a", 1, max_judges + 1);
+		assert_noop!(
+			ContractsRegistry::add_judge(RawOrigin::Root.into(), acc),
+			Error::<Test>::TooMany
+		);
+	});
+}
+
+#[test]
+fn judge_can_not_sign_not_existing_contract() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let acc = account("a", 1, SEED);
+
+		assert_ok!(ContractsRegistry::add_judge(origin, acc));
+
+		let origin = RawOrigin::Signed(acc);
+		assert_noop!(
+			ContractsRegistry::judge_sign_contract(origin.into(), 0),
+			Error::<Test>::ContractNotFound
+		);
+	});
+}
+
+#[test]
+fn judge_can_sign_existing_contract() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let acc = account("a", 1, SEED);
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			RuntimeOrigin::signed(2),
+			vec![3, 4, 5].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 95u64);
+
+		assert_ok!(ContractsRegistry::add_judge(origin, acc));
+
+		let origin = RawOrigin::Signed(acc);
+		assert_ok!(ContractsRegistry::judge_sign_contract(origin.into(), 0));
+	});
+}
+
+#[test]
+fn only_judge_can_use_judge_sign_contract() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::signed(2);
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			origin.clone().into(),
+			vec![3, 4, 5].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 95u64);
+
+		assert_noop!(
+			ContractsRegistry::judge_sign_contract(origin.into(), 0),
+			Error::<Test>::NotJudge
+		);
+	});
+}
+
+#[test]
+fn judge_already_signed_contract() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::root();
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			RuntimeOrigin::signed(2),
+			vec![3, 4, 5].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 95u64);
+
+		let acc = account("a", 1, SEED);
+		assert_ok!(ContractsRegistry::add_judge(origin, acc));
+
+		let origin = RawOrigin::Signed(acc);
+		assert_ok!(ContractsRegistry::judge_sign_contract(origin.clone().into(), 0));
+
+		assert_noop!(
+			ContractsRegistry::judge_sign_contract(origin.into(), 0),
+			Error::<Test>::AlreadySigned
+		);
+	});
+}
+
+#[test]
+fn party_can_sign_contract() {
```

Comment:
Missing tests for party signatures for:
* non-existent contract
* origin not being a party to the contract